### PR TITLE
Create utillinuxMinimal and utillinux (full).

### LINKED ISSUE #11637: Build util-linux with systemd support.
This is more of a question the a bug, but it would be nice to find the best solution.

Some util-linux programs have extra features that are available when compiled with systemd support. For example `logger --journald` can be used to add structured data into the journal from the shell.

The hard part is that in nixos systemd depends on util-linux which creates a dependency cycle so it isn't as simple as adding `systemd pkgconfig` to the requirements.

Below I have included a workaround that creates a new package with systemd support, however this isn't particularly nice because it means that you have to have two versions of util-linux.

``` nix
{
    nixpkgs.config.packageOverrides = pkgs: rec {
        utillinux-sd = pkgs.stdenv.lib.overrideDerivation pkgs.utillinux (old: {
        nativeBuildInputs = old.nativeBuildInputs ++ [ pkgs.systemd pkgs.pkgconfig ];
            configureFlags = old.configureFlags + ''
                --with-systemd
                --with-systemdsystemunitdir=$out/lib/systemd/system/
            '';
        });
    };
}
```

The `--with-systemd` flag isn't required but it is nice because it causes a failure if systemd isn't found.

`--with-systemdsystemunitdir` is required otherwise it tries to write it's unit file into the systemd store directory.

### PR DESCRIPTION
**Pull request by kevincox**
This is the start of the change to fix #11637. Currently I am having problems with circular dependencies as it appears that system is still pulling in a utillinux with a non-null `systemd` argument even though I have told it to use `utillinuxMinimal`.

### CODE REVIEW

File: pkgs/os-specific/linux/util-linux/default.nix [hunk 1]
```diff
@@ -1,7 +1,7 @@
-{ stdenv, fetchurl, zlib, ncurses ? null, perl ? null, pam }:
+{ stdenv, fetchurl, zlib, ncurses ? null, perl ? null, pam, systemd ? null }:
 
 stdenv.mkDerivation rec {
-  name = "util-linux-2.27.1";
+  name = "util-linux-2.27.1${if systemd == null then "-minimal" else "-full"}";
```
Comment (kevincox) [inline pkgs/os-specific/linux/util-linux/default.nix]: This is just for debugging.

**Comment by kevincox** [comment]
Another question is should I change every other reference to minimal or add curses and systemd support to them?

**Comment by kevincox** [comment]
Also note that this is never going to pass on travis because the number of packages that need to be rebuilt is quite large.

**Comment by vcunat** [comment]
I think it's good. I staged it now.

**Pull request closed without merging**